homeassistant/components/vicare/switch.py

```diff
@@ -0,0 +1,211 @@
+        self.entity_description = description
+        self._device_config = device_config
+        self._api = api
+        self._ignore_update_until = datetime.datetime.utcnow()
+        self._state = None
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        return self._state
+
+    async def async_update(self):
+        """update internal state"""
+        now = datetime.datetime.utcnow()
+        """we have identified that the API does not directly sync the represented state, therefore we want to keep
+        an assumed state for a couple of seconds - so lets ignore an update
+        """
+        if now < self._ignore_update_until:
+            _LOGGER.debug(
+                "Ignoring Update Request for OneTime Charging for some seconds"
+            )
+            return
+
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                _LOGGER.debug("Fetching DHW One Time Charging Status")
+                self._state = await self.hass.async_add_executor_job(
+                    self.entity_description.value_getter, self._api
+                )
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("Unable to retrieve data from ViCare server")
+        except ValueError:
+            _LOGGER.error("Unable to decode data from ViCare server")
+        except PyViCareRateLimitError as limit_exception:
+            _LOGGER.error("Vicare API rate limit exceeded: %s", limit_exception)
+        except PyViCareInvalidDataError as invalid_data_exception:
+            _LOGGER.error("Invalid data from Vicare server: %s", invalid_data_exception)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Handle the button press."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                """Turn the switch on."""
+                _LOGGER.debug("Enabling DHW One-Time-Charging")
+                await self.hass.async_add_executor_job(
+                    self.entity_description.enabler, self._api
+                )
+                self._ignore_update_until = (
+                    datetime.datetime.utcnow() + TIMEDELTA_UPDATE
+                )
+                self._state = True
+
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("Unable to retrieve data from ViCare server")
+        except ValueError:
+            _LOGGER.error("Unable to decode data from ViCare server")
+        except PyViCareRateLimitError as limit_exception:
+            _LOGGER.error("Vicare API rate limit exceeded: %s", limit_exception)
+        except PyViCareInvalidDataError as invalid_data_exception:
+            _LOGGER.error("Invalid data from Vicare server: %s", invalid_data_exception)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Handle the button press."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
```

Comment:
If the command is not supported, it should not be suppressed, the error should be propagated